src/modules/hip/kernel/non_silent_region_detection.hpp

```diff
@@ -0,0 +1,410 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+// -------------------- Set 0 -  moving mean square kernel device helpers --------------------
+
+// calculate the position in shared memory to avoid bank conflicts
+__host__ __device__ __forceinline__ int smem_pos(int pos)
+{
+    return pos + (pos >> 5); // since shared memory banks considered is 32
+}
+
+__device__ __forceinline__ void compute_prefix_sum(float *input, uint bufferLength)
+{
+    int offset = 1;
+    int offsetA = offset - 1;
+    int offsetB = 2 * offset - 1;
+    int tid = hipThreadIdx_x;
+    int blockDimMul2 = 2 * hipBlockDim_x;
+
+    for (int d = bufferLength >> 1; d > 0; d >>= 1)
+    {
+        __syncthreads();
+        int dMul2 = 2 * d;
+        for (int idxMul2 = 2 * tid; idxMul2 < dMul2; idxMul2 += blockDimMul2)
+            input[smem_pos(offset * idxMul2 + offsetB)] += input[smem_pos(offset * idxMul2 + offsetA)];
+
+        offset <<= 1;
+        offsetA = offset - 1;
+        offsetB = 2 * offset - 1;
+    }
+
+    if (tid == 0)
+    {
+        int last = bufferLength - 1;
+        input[smem_pos(last)] = 0;
+    }
+
+    for (int d = 1; d < bufferLength; d <<= 1)
+    {
+        offset >>= 1;
+        offsetA = offset - 1;
+        offsetB = 2 * offset - 1;
+        __syncthreads();
+
+        int dMul2 = 2 * d;
+        for (int idxMul2 = 2 * tid; idxMul2 < dMul2; idxMul2 += blockDimMul2)
+        {
+            int smem_posA = smem_pos(offset * idxMul2 + offsetA);
+            int smem_posB = smem_pos(offset * idxMul2 + offsetB);
```

Comment:
The protocol on this naming discussed was to use camelCase_smem only if the variable is shared memory. Naming functions and local variables as smem too is a little confusing. Maybe call the function "compute_position_in_smem()"

Also suggest vectorizing variables to pos_i2.x and pos_i2.y, and use an offset_i2 instead of offsetA and offsetB